From 97a6b6e2611c3246aadaa7d8ea5ec45262afb900 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Fri, 5 Sep 2014 11:03:53 -0400
Subject: [ethernet] sfc: Fail self-test with -EBUSY, not -EIO, if the device is busy

Message-id: <1409915048-26438-18-git-send-email-nikolay@redhat.com>
Patchwork-id: 91419
O-Subject: [RHEL7.1 PATCH BZ 1110888 17/32] sfc: Fail self-test with -EBUSY, not -EIO, if the device is busy
Bugzilla: 1110888
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Ben Hutchings <bhutchings@solarflare.com>

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Signed-off-by: Shradha Shah <sshah@solarflare.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 5eed1f68523c8cdeeafae24350200d47ddc293b7)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/sfc/ethtool.c b/drivers/net/ethernet/sfc/ethtool.c
index 3444dd6..7d5cb86 100644
--- a/drivers/net/ethernet/sfc/ethtool.c
+++ b/drivers/net/ethernet/sfc/ethtool.c
@@ -461,7 +461,7 @@ static void efx_ethtool_self_test(struct net_device *net_dev,
   goto fail;
 
  if (efx->state != STATE_READY) {
-  rc = -EIO;
+  rc = -EBUSY;
   goto out;
  }
 
-- 
1.7.1